tracer/test/Datadog.Trace.Tests/DatabaseMonitoring/DatabaseMonitoringPropagatorTests.cs

```diff
@@ -112,13 +114,58 @@ public void ExpectedCommentInjectedV1()
             span.SetTraceSamplingPriority(samplingPriority);
 
-            var returnedComment = DatabaseMonitoringPropagator.PropagateSpanData(dbmPropagationLevel, "Test.Service", "MyDatabase", "MyHost", span, integrationId, out var traceParentInjectedValue);
+            var returnedComment = DatabaseMonitoringPropagator.PropagateDataViaComment(dbmPropagationLevel, "Test.Service", "MyDatabase", "MyHost", span, integrationId, out var traceParentInjectedValue);
 
             traceParentInjectedValue.Should().Be(traceParentInjected);
             returnedComment.Should().Be(expectedComment);
         }
+
+        [Theory]
+        [InlineData("full", "sqlclient", SamplingPriorityValues.UserKeep, true, "01BEEFBEEFBEEFBEEFBABEBABEBABEBABECAFECAFECAFECAFE")]
+        [InlineData("full", "sqlclient", SamplingPriorityValues.UserReject, true, "00BEEFBEEFBEEFBEEFBABEBABEBABEBABECAFECAFECAFECAFE")]
+        [InlineData("nope", "sqlclient", SamplingPriorityValues.UserKeep, false, null)]
+        // disabled for all db types except sqlclient for now
+        [InlineData("full", "npgsql", SamplingPriorityValues.UserKeep, false, null)]
+        [InlineData("full", "sqlite", SamplingPriorityValues.UserKeep, false, null)]
+        [InlineData("full", "oracle", SamplingPriorityValues.UserKeep, false, null)]
+        [InlineData("full", "mysql", SamplingPriorityValues.UserKeep, false, null)]
+        public void ExpectedContextSet(string propagationMode, string integration, int? samplingPriority, bool shouldInject, string expectedContext)
+        {
+            Enum.TryParse(propagationMode, ignoreCase: true, out DbmPropagationLevel dbmPropagationLevel);
+            Enum.TryParse(integration, ignoreCase: true, out IntegrationId integrationId);
+
+            // capture command and parameter sent
+            string sql = null;
+            byte[] context = null;
+            var connectionMock = new Mock<IDbConnection>(MockBehavior.Strict);
+            var commandMock = new Mock<IDbCommand>();
+            var parameterMock = new Mock<IDbDataParameter>();
+            connectionMock.Setup(c => c.CreateCommand()).Returns(commandMock.Object);
+            commandMock.SetupSet(c => c.CommandText = It.IsAny<string>())
+                       .Callback<string>(value => sql = value);
+            commandMock.Setup(c => c.CreateParameter()).Returns(parameterMock.Object);
+            commandMock.SetupGet(c => c.Parameters).Returns(Mock.Of<IDataParameterCollection>());
+            parameterMock.SetupSet(p => p.Value = It.IsAny<byte[]>())
+                         .Callback<object>(value => context = (byte[])value);
+
+            var span = _v0Tracer.StartSpan("db.query", parent: SpanContext.None, serviceName: "pouet", traceId: new TraceId(Upper: 0xBABEBABEBABEBABE, Lower: 0xCAFECAFECAFECAFE), spanId: 0xBEEFBEEFBEEFBEEF);
+            span.SetTraceSamplingPriority((SamplingPriority)samplingPriority.Value);
```

Comment:
It looks like the `samplingPriority` parameters doesn't need to be nullable (which lets you remove the `.Value` here).

Also, the `span.SetTraceSamplingPriority()` extension method is a legacy-ish public API that uses the deprecated-ish `SamplingPriority` enum. Internally, we want to avoid using that and use `TraceContext.SetSamplingPriority(int)` instead.

```csharp
span.Context.TraceContext.SetSamplingPriority(samplingPriority);
```

Reply:
The rule of thumb is to use `int` and the `SamplingPriorityValues` constants instead of the `SamplingPriority` enum.